src/main/java/com/jcraft/jsch/Session.java

```diff
@@ -2718,15 +2719,45 @@ public void setTimeout(int timeout) throws JSchException {
   public String getServerVersion(){
     return Util.byte2str(V_S);
   }
+  
+  /**
+   * Returns the client version string used during an active connection.
+   * If no connection is active <code>null</code> is returned.
+   * @return The used version string or <code>null</code> if no
+   * connection is active.
+   */
+  public String getUsedClientVersion() {
```

Comment:
I personally find this extremely convoluted and would really rather we not add this method.

Reply:
Since two members are necessary to prevent the effect you've described I though, offering a get-method to allow access to these two members should be available. But I have no special feelings about this method, so I've removed it.